cpp/dolfinx/io/checkpointing.cpp

```diff
@@ -0,0 +1,354 @@
+// Copyright (C) 2024 Abdullah Mujahid
+//
+// This file is part of DOLFINX (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#ifdef HAS_ADIOS2
+
+#include "checkpointing.h"
+#include <adios2.h>
+#include <basix/finite-element.h>
+#include <dolfinx/fem/CoordinateElement.h>
+#include <dolfinx/mesh/Mesh.h>
+#include <dolfinx/mesh/utils.h>
+#include <mpi.h>
+
+/// @file checkpointing.h
+/// @brief ADIOS2 based checkpointing
+namespace
+{
+std::map<basix::element::lagrange_variant, std::string> variant_to_string{
+    {basix::element::lagrange_variant::unset, "unset"},
+    {basix::element::lagrange_variant::equispaced, "equispaced"},
+    {basix::element::lagrange_variant::gll_warped, "gll_warped"},
+};
```

Comment:
```suggestion
std::map<basix::element::lagrange_variant, std::string> variant_to_string{
    {basix::element::lagrange_variant::unset, "unset"},
    {basix::element::lagrange_variant::equispaced, "equispaced"},
    {basix::element::lagrange_variant::gll_warped, "gll_warped"},
};
```
There should potentially be a wrapper for this in Basix. @mscroggs  what do you think?

Reply:
Think it used to exist but we removed it... or maybe it only existed in the Python wrapper